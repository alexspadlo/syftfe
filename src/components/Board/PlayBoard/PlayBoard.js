import React from 'react'
import { useDispatch } from 'react-redux'
import { selectCell } from '../../../store/actions/moves'
import tictocImage from '../../../static/tictoctoe.png'
import PropTypes, { any } from 'prop-types'
import s from './PlayBoard.module.scss'

const PlayBoard = ({ currentPlayer, board }) => {
    const dispatch = useDispatch()
    if(!board || !currentPlayer) return null
    return (
        <section className={`w-100 ${s.containerBack}`}>
            <div className={`container`}>
                <div className="row m-0">
                    <div className={`col-12 col-lg-5 m-auto ${s.customPaddingTop} text-center`}>
                        <img className="img-fluid" src={tictocImage} alt="tictoc" />
                    </div>
                    <div className={`col-12 col-lg-7 ${s.customPaddingTop} m-auto`}>
                        <div className="row m-0">
                            {board && board.length > 0 &&
                                board.map((row, rowIndex) => (
                                    <div key={`cvUnique${rowIndex}`} className="col-lg-9 col-12 m-auto">
                                        <div className="row m-0">
                                            {row && row.length > 0 &&
                                                row.map((colValue, colIndex) => (
                                                    <div
                                                        onClick={() => {
                                                            if (!colValue) {
                                                                dispatch(selectCell(currentPlayer, rowIndex, colIndex))
                                                            }
                                                        }}
                                                        key={`boxUnique${colIndex}`}
                                                        className={`col ${s.clickText} ${colValue ? s.noCursor : ''} ${s.tictok} ${s[`box${rowIndex}col${colIndex}`]}`}>
                                                        {colValue ? colValue : ''}
                                                    </div>
                                                ))}
                                        </div>
                                    </div>
                                ))}
                        </div>
                    </div>
                </div>
            </div>
        </section>
    )
}

PlayBoard.propTypes = {
    currentPlayer: PropTypes.string,
    board: PropTypes.arrayOf(PropTypes.arrayOf(any))
}

export default PlayBoard