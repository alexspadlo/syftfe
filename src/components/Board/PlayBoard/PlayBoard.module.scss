.containerBack {
  background-size: cover;
  background-color: rgb(160, 81, 136);
  min-height: 100vh;
  height: 100%;

  .customPaddingTop {
    padding-top: 120px;
    padding-bottom: 100px;

    .textPic {
      min-height: 455px;
    }

    .clickText {
      color: white;
      text-align: center;
      font-size: 80px;
      padding: 10px;
    }

    .tictok {
      height: 150px;
      width: 150px;
      float: left;
      border: 3px solid white;
      cursor: pointer;

      &.box0col0 {
        border-left: none;
        border-top: none;
      }

      &.box0col1 {
        border-top: none;
      }

      &.box0col2 {
        border-right: none;
        border-top: none;
      }

      &.box1col0 {
        border-left: none;
      }

      &.box1col1 {
        border: 3px solid white;
      }

      &.box1col2 {
        border-right: none;
      }

      &.box2col0 {
        border-left: none;
        border-bottom: none;
      }

      &.box2col1 {
        border-bottom: none;
      }

      &.box2col2 {
        border-right: none;
        border-bottom: none;
      }

      &.noCursor {
        cursor: default;
      }
    }
  }
}
