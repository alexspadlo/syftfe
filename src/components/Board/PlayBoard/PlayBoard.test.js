import React from 'react'
import { render } from '@testing-library/react'
import PlayBoard from './PlayBoard'
import configureStore from '../../../store'
import { Provider } from 'react-redux'

const store = configureStore()

const board = [
	[null, null, 'X'],
	[null, 'X', null],
	[null, null, 'O']
]

const currentPlayer = 'X'

const fullProps = {
	board, 
	currentPlayer
}


describe('Testing PlayBoard Component', () => {
	test('Do not render PlayBoard without the props', () => {
		const { container } = render(
			<Provider store={store}>
				<PlayBoard />
			</Provider>
		)
		expect(container.firstChild).toBeNull()
	})
	test('Should not render without the currentplayer props', () => {
		const { container } = render(
			<Provider store={store}>
				<PlayBoard board={board} />
			</Provider>
		)
		expect(container.firstChild).toBeNull()
	})
	test('Should not render without the board props', () => {
		const { container } = render(
			<Provider store={store}>
				<PlayBoard currentPlayer={currentPlayer} />
			</Provider>
		)
		expect(container.firstChild).toBeNull()
	})
	test('Render with fullprops has 3 noCursor class and match snapshot', () => {
		const { container } = render(
			<Provider store={store}>
				<PlayBoard {...fullProps} />
			</Provider>
		)
		const findNoCursorclass = container.querySelectorAll('.noCursor')
		expect(findNoCursorclass).toHaveLength(3)
		expect(container.firstChild).toMatchSnapshot()
	})
})