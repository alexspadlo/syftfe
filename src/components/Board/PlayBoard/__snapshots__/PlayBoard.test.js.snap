// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PlayBoard Component Render with fullprops has 3 noCursor class and match snapshot 1`] = `
<section
  class="w-100 containerBack"
>
  <div
    class="container"
  >
    <div
      class="row m-0"
    >
      <div
        class="col-12 col-lg-5 m-auto customPaddingTop text-center"
      >
        <img
          alt="tictoc"
          class="img-fluid"
          src="tictoctoe.png"
        />
      </div>
      <div
        class="col-12 col-lg-7 customPaddingTop m-auto"
      >
        <div
          class="row m-0"
        >
          <div
            class="col-lg-9 col-12 m-auto"
          >
            <div
              class="row m-0"
            >
              <div
                class="col clickText  tictok box0col0"
              />
              <div
                class="col clickText  tictok box0col1"
              />
              <div
                class="col clickText noCursor tictok box0col2"
              >
                X
              </div>
            </div>
          </div>
          <div
            class="col-lg-9 col-12 m-auto"
          >
            <div
              class="row m-0"
            >
              <div
                class="col clickText  tictok box1col0"
              />
              <div
                class="col clickText noCursor tictok box1col1"
              >
                X
              </div>
              <div
                class="col clickText  tictok box1col2"
              />
            </div>
          </div>
          <div
            class="col-lg-9 col-12 m-auto"
          >
            <div
              class="row m-0"
            >
              <div
                class="col clickText  tictok box2col0"
              />
              <div
                class="col clickText  tictok box2col1"
              />
              <div
                class="col clickText noCursor tictok box2col2"
              >
                O
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
